rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1039,23 +1036,106 @@ public int getBtcBlockchainInitialBlockHeight() throws IOException {
+
+            // Block not found, default to basic cost
+            if (block == null) {
+                return BASIC_COST;
+            }
+
+            final int bestChainHeight = getBtcBlockchainBestChainHeight();
+
+            // Make sure calculated depth is >= 0
+            final int blockDepth = Math.max(0, bestChainHeight - block.getHeight());
+
+            // Block too deep, default to basic cost
+            if (blockDepth > BTC_TRANSACTION_CONFIRMATION_MAX_DEPTH) {
+                return BASIC_COST;
+            }
+
+            return BASIC_COST + blockDepth*STEP_COST;
+        } catch (IOException | BlockStoreException e) {
+            logger.warn("getBtcTransactionConfirmationsGetCost btcBlockHash:{} there was a problem " +
+                    "gathering the block depth while calculating the gas cost. " +
+                    "Defaulting to basic cost.", btcBlockHash, e);
+            return BASIC_COST;
         }
-        return current.getHeader().getHash();
+    }
+
+    /**
+     * @param btcTxHash The BTC transaction Hash
+     * @param btcBlockHash The BTC block hash
+     * @param merkleBranch The merkle branch
+     * @throws BlockStoreException
+     * @throws IOException
+     */
+    public Integer getBtcTransactionConfirmations(Sha256Hash btcTxHash, Sha256Hash btcBlockHash, MerkleBranch merkleBranch) throws BlockStoreException, IOException {
+        Context.propagate(btcContext);
+        this.ensureBtcBlockChain();
+
+        // Get the block using the given block hash
+        StoredBlock block = btcBlockStore.getFromCache(btcBlockHash);
+        if (block == null) {
+            return BTC_TRANSACTION_CONFIRMATION_INEXISTENT_BLOCK_HASH_ERROR_CODE;
+        }
+
+        final int bestChainHeight = getBtcBlockchainBestChainHeight();
+
+        // Prevent diving too deep in the blockchain to avoid high processing costs
+        final int blockDepth = Math.max(0, bestChainHeight - block.getHeight());
+        if (blockDepth > BTC_TRANSACTION_CONFIRMATION_MAX_DEPTH) {
+            return BTC_TRANSACTION_CONFIRMATION_BLOCK_TOO_OLD_ERROR_CODE;
+        }
+
+        try {
+            StoredBlock storedBlock = btcBlockStore.getStoredBlockAtMainChainHeight(block.getHeight());
+            // Make sure it belongs to the best chain
+            if (storedBlock == null || !storedBlock.equals(block)){
+                return BTC_TRANSACTION_CONFIRMATION_BLOCK_NOT_IN_BEST_CHAIN_ERROR_CODE;
+            }
+        } catch (BlockStoreException e) {
+            logger.warn(String.format(
+                    "Illegal state trying to get block with hash {}",
+                    btcBlockHash
+            ), e);
+            return BTC_TRANSACTION_CONFIRMATION_INCONSISTENT_BLOCK_ERROR_CODE;
+        }
+
+        if (!merkleBranch.proves(btcTxHash, block.getHeader())) {
+            return BTC_TRANSACTION_CONFIRMATION_INVALID_MERKLE_BRANCH_ERROR_CODE;
+        }
+
+        return bestChainHeight - block.getHeight() + 1;
```

Comment:
Currently we are following @SergioDemianLerner  idea of returning negative numbers in precompiled contracts instead of throwing an error. Each number corresponds to an unique error type.